paseos/tests/activity_test.py

```diff
@@ -9,15 +13,112 @@ def test_activity():
     # Initial power is 500
     assert sat1.battery_level_in_Ws == 500
 
+    # Out test case is a function that increments a value, genius.
+    # (needs a list so increase the actual value by reference and don't create a copy)
+    test_value = [0]
+
+    async def func(args):
+        for _ in range(10):
+            args[0][0] += 1
+            # print(args)
+            await asyncio.sleep(0.2)
+
     # Register an activity that draws 10 watt per second
-    sim.register_activity("Testing", power_consumption_in_watt=10)
+    sim.register_activity(
+        "Testing", activity_function=func, power_consumption_in_watt=10
+    )
 
-    # Run the activity for 5 seconds
-    sim.perform_activity("Testing", duration_in_s=5)
+    # Run the activity
+    sim.perform_activity("Testing", activity_func_args=[test_value])
+
+    # Check activity result
+    assert test_value[0] == 10
 
     # Check power was depleted as expected
-    assert sat1.battery_level_in_Ws == 450
+    # Activity should run roughly 2s
+    # We charge 1W per second
+    # And discharge 10W per second
+    # So should be roughly 20W - 6W consumed from starting 500
+    assert sat1.battery_level_in_Ws > 480 and sat1.battery_level_in_Ws < 490
+
+
+def test_running_two_activities():
+    """This test ensures that you cannot run two activities at the same time."""
+    sim, sat1, earth = get_default_instance()
+
+    # Declare two activities, one calls the other
+    test_value = [0]
+
+    async def act1(args):
+        args[0][0] = 42  # this gets executed
+        sim.perform_activity("act2")
+        await asyncio.sleep(0.1)
+
+    async def act2(args):
+        args[0][0] = -1  # this won't
+        await asyncio.sleep(0.1)
+
+    # Register both activities
+    sim.register_activity("act1", activity_function=act1, power_consumption_in_watt=10)
+    sim.register_activity("act2", activity_function=act2, power_consumption_in_watt=10)
+
+    # try running it
+    sim.perform_activity("act1", activity_func_args=[test_value])
```

Comment:
It would be nice if there is a stream-lined way to obtain the output of an activity instead of overwriting the input arguments.

Reply:
Added to #42 as further improvements. Would not treat here now so that we can first focus on improving our docs